### PR TITLE
fix some problems in proxy

### PR DESCRIPTION
**Pull request by ikaven1024**
**What type of PR is this?**

/kind api-change
/kind bug

**What this PR does / why we need it**:
1. Fix proxy get request will modify data in cache
2. change the error when creating member cluster resource.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
When creating resource defined in ResourceRegistry, `MethodNotSupported` error will be responsed
```

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks!
/lgtm

In fact, we don't change the API in this pr.
/remove-kind api-change

Website docs is updated by https://github.com/karmada-io/website/pull/140 at the same time.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

File: pkg/search/proxy/store/multi_cluster_cache.go [hunk 1]
```diff
@@ -146,15 +146,16 @@ func (c *MultiClusterCache) Get(ctx context.Context, gvr schema.GroupVersionReso
...
+	accessor, err := meta.Accessor(cloneObj)
 	if err != nil {
 		return nil, err
 	}
 
 	mrv := newMultiClusterResourceVersionWithCapacity(1)
 	mrv.set(clusterName, accessor.GetResourceVersion())
 	accessor.SetResourceVersion(mrv.String())
-	return obj, err
+	return cloneObj, err
```
Comment (liblikewhen) [inline pkg/search/proxy/store/multi_cluster_cache.go:158]: I think you don't need copy it every time when call the get method,  just copy it whenever need to modify the object

Comment (ikaven1024) [reply pkg/search/proxy/store/multi_cluster_cache.go:158]: `accessor.SetResourceVersion(mrv.String())`  it is just modifying it.

Comment (liblikewhen) [reply pkg/search/proxy/store/multi_cluster_cache.go:158]: so, why not deepcopy the object when call the accessor.SetResourceVersion(mrv.String())

Comment (ikaven1024) [reply pkg/search/proxy/store/multi_cluster_cache.go:158]: > so, why not deepcopy the object when call the accessor.SetResourceVersion(mrv.String())

Do you mean as below? I don't think it has too much different. 
```go
	mrv := newMultiClusterResourceVersionWithCapacity(1)
	mrv.set(clusterName, accessor.GetResourceVersion())

	cloneObj := obj.DeepCopyObject()
	accessor, err := meta.Accessor(cloneObj)
	if err != nil {
		return nil, err
	}
	accessor.SetResourceVersion(mrv.String())
	return cloneObj, err
```

Comment (liblikewhen) [reply pkg/search/proxy/store/multi_cluster_cache.go:158]: yeah. I think it will reduce the memory copy and save some memory

Comment (ikaven1024) [reply pkg/search/proxy/store/multi_cluster_cache.go:158]: In fact, err will not occur normally, and copy will be always done

**Pull request merged**